tests/testthat/test-hcai-impute.R

```diff
@@ -172,16 +173,13 @@ test_that("bag imputation bakes expected results", {
   expect_equal(d_imputed$length[14], 7.797, tolerance = 2)
 })
 
-test_that("random columns don't get imputed", {
-  # If a categorical column is completely uncorrelated with other columns, bag
-  # impute will not fill in the missing data. This test is mostly just to make
-  # sure this function is behaving as it was when it was written.
-
+test_that("random columns get imputed when factors", {
   # add randomly distributed columns with NAs
   df$random_chars <- sample(c("NYC", "Chicago"), n, replace = TRUE)
   df$random_nums <- sample(1:n, n, replace = FALSE)
   df$random_chars[sample(1:n, 55, replace = FALSE)] <- NA
   df$random_nums[sample(1:n, 45, replace = FALSE)] <- NA
+  df$random_chars <- as.factor(df$random_chars)
```

Comment:
This comment was that way because we didn't know it was a character/factor thing. In random character columns, `collapse_rare_factors` didn't hide the error, but I never made the connection. Great work!